src/electronic-ids/pkcs11/Pkcs11ElectronicID.cpp

```diff
@@ -149,11 +149,19 @@ ElectronicID::PinRetriesRemainingAndMax Pkcs11ElectronicID::authPinRetriesLeft()
 pcsc_cpp::byte_vector Pkcs11ElectronicID::signWithAuthKey(const pcsc_cpp::byte_vector& pin,
                                                           const pcsc_cpp::byte_vector& hash) const
 {
-    validateAuthHashLength(authSignatureAlgorithm(), name(), hash);
-
-    const auto signature = manager.sign(authToken, hash, authSignatureAlgorithm().hashAlgorithm(),
-                                        reinterpret_cast<const char*>(pin.data()), pin.size());
-    return signature.first;
+    try {
+        validateAuthHashLength(authSignatureAlgorithm(), name(), hash);
+
+        const auto signature =
+            manager.sign(authToken, hash, authSignatureAlgorithm().hashAlgorithm(),
+                         reinterpret_cast<const char*>(pin.data()), pin.size());
+        return signature.first;
+    } catch (const VerifyPinFailed& e) {
+        if (e.status() == VerifyPinFailed::Status::RETRY_ALLOWED && module.retryMax == -1) {
+            throw VerifyPinFailed(VerifyPinFailed::Status::RETRY_ALLOWED, nullptr, -1);
```

Comment:
As we discussed, I would ideally see this case resolved inside `PKCS11CardManager`, but you correctly pointed out that you would need to pass `module.retryMax` into it as a parameter which would be inconsistent as we don't do that with other parameters. I'll add a comment to clarify this and merge this as-is, but there might be room for improvement here.